functions/__ssh_agent_is_started.fish

```diff
@@ -7,6 +7,8 @@ function __ssh_agent_is_started -d "check if ssh agent is already started"
 		return 1
 	end
 
-	ps -ef | grep $SSH_AGENT_PID | grep -v grep | grep -q ssh-agent
-	return $status
+	ssh-add -l > /dev/null
+	if [ $status -eq 2 ]
```

Comment:
Let's stick with `test ...` instead of `[ ... ]`, as it's used in other places in the code.